src/functional/ffront/foast_passes/type_deduction.py

```diff
@@ -509,6 +509,12 @@ def _visit_where(self, node: foast.Call, **kwargs) -> foast.Call:
 
         try:
             return_type = type_info.promote(left_type, right_type)
+
+            if isinstance(mask_type, ct.FieldType) and isinstance(return_type, ct.ScalarType):
+                return_type = type_info.promote(
+                    return_type, ct.FieldType(dims=mask_type.dims, dtype=return_type)
+                )
+
```

Comment:
just
```suggestion
            return_type = type_info.promote(mask_type, left_type, right_type)
```
should probably work

Reply:
That change results in this exception, probably as the dtype in `mask_type` is bool, and the others are float. 
```bash
   raise GTTypeError("Could not promote scalars of different dtype (not implemented).")
E   functional.common.GTTypeError: Could not promote scalars of different dtype (not implemented).
```

Reply:
oh that makes sense